detection_rules/rule_validators.py

```diff
@@ -73,6 +73,29 @@ def text_fields(self, eql_schema: ecs.KqlSchema2Eql) -> List[str]:
     def unique_fields(self) -> List[str]:
         return list(set(str(f) for f in self.ast if isinstance(f, eql.ast.Field)))
 
+    def validate_query_with_schema(self, schema: Union[ecs.KqlSchema2Eql, endgame.EndgameSchema],
+                                   err_trailer: str, beat_types: list = None) -> None:
+        """Validate the query against the schema."""
+        try:
+            with schema, eql.parser.elasticsearch_syntax, eql.parser.ignore_missing_functions:
+                eql.parse_query(self.query)
+        except eql.EqlParseError as exc:
+            message = exc.error_msg
+            trailer = err_trailer
+            if "Unknown field" in message and beat_types:
+                trailer = f"\nTry adding event.module or event.dataset to specify beats module\n\n{trailer}"
+            elif "Field not recognized" in message:
+                text_fields = self.text_fields(schema)
+                if text_fields:
+                    fields_str = ', '.join(text_fields)
```

Comment:
Good catch. It's because of youre other [comment](https://github.com/elastic/detection-rules/pull/2257#discussion_r997547337). I added a check to properly get the schema information.